clusterctl/clusterdeployer/clusterdeployer.go

```diff
@@ -60,6 +60,7 @@ type ClusterClient interface {
 	CreateMachineSetObjects([]*clusterv1.MachineSet) error
 	CreateMachineObjects([]*clusterv1.Machine) error
 	DeleteClusterObjects() error
+	DeleteApiServerStackDeployments() error
```

Comment:
Why is awareness of API stack added to the client? There is no CreateApiServerStackDeployements() in the client. It seems it would make sense to be more consistent with the scope of awareness and keep the awareness only in clusterdeployer.